codec-http/src/main/java/io/netty/handler/codec/http/websocketx/extensions/compression/WebSocketClientCompressionHandler.java

```diff
@@ -27,12 +27,14 @@
 @ChannelHandler.Sharable
 public final class WebSocketClientCompressionHandler extends WebSocketClientExtensionHandler {
 
+    private static final int MAX_ALLOCATION = 1024 * 1024 * 10;
```

Comment:
Why was 10 MiB chosen as the limit? Is this something people might want to make configurable?

Reply:
I think we should deprecate the `public static final WebSocketClientCompressionHandler INSTANCE = new WebSocketClientCompressionHandler();` and then just add another constructor that takes the max allocation size as the constructor 
`